### DISCUSSION TITLE
CosmWasm Support for Nillion

### DISCUSSION THREAD

**Discussion by envoy1084** [NillionNetwork/discussions#74] [Ideas]
Hi I was wondering if there would be CosmWasm support for Nillion in the future.

So I was searching for nillion chain config for connecting web cosmos wallet and came across this

https://github.com/cosmos/chain-registry/pull/4734

## Why CosmWasm?

So basically current scenario we are able to save secrets and compute using programs. But to do these thing we need to sign with a wallet again and again which degrades UX.

CosmWasm could provide a great way to open Nillion to programatic access and interactions with multiple chains, with some researching I got this projects on Cosmos which intergrate with other blockchains

1. Aura's Implementation of Smart Account on Cosmos: https://github.com/aura-nw/aura/tree/main/x/smartaccount This could significantly improve UX with programmable EOA and we could generate Payment Quotes based on certain on-chain events or even cross-chain ones.
2. Account Abstract: https://github.com/larry0x/abstract-account These are smart accounts which are controlled by Ethereum Wallets or even Signatures, so for instance in the future we could generate signatures privately and execute transactions on CosmWasm

**Comment by wwwehr** [comment]
@Davetbutler FYI

@Envoy-VC thanks again for writing this up!
